pmsoService/product_flow/features/orders/app/validators.py

```
@@ -1,36 +1,76 @@
 # ------------------------------- Sale Order Fields Validators ---------------------------------------
 def validate_sale_order(due_date: date):
     if due_date < date.today():
         raise ValueError("Due date cannot be in the past")
 
+
 def validate_order_item(item: CreateSaleOrderItem):
-    if bool(item.product_id) == bool(item.material_id):
-        raise ValueError("Each item must have a unique product ID")
-    if item.quantity <= 0 :
+    if not item.product_id and not item.material_id:
+        raise ValueError("Each item must have either a product_id, material_id, or both")
```

Ruff (0.14.2)

Avoid specifying long messages outside the exception class

(TRY003)